File: OSX Examples/Objective-C/Convolution/Convolution/ViewController.m
//
//  ViewController.m
//  Convolution
//
//  Created by Aurelius Prochazka on 2/16/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

#import "ViewController.h"
#import "AKFoundation.h"
#import "AKTools.h"
#import "ConvolutionInstrument.h"

@implementation ViewController
{
    ConvolutionInstrument *conv;
}


- (void)viewDidLoad {
    [super viewDidLoad];
    conv = [[ConvolutionInstrument alloc] init];
    [AKOrchestra addInstrument:conv];
    [AKOrchestra start];
}

- (IBAction)start:(id)sender {
    [conv play];
}

- (IBAction)stop:(id)sender {
    [conv stop];
}

- (IBAction)changeDryWet:(id)sender {
    [AKTools setProperty:conv.dryWetBalance withSlider:(NSSlider *)sender];
}
- (IBAction)changeDishWell:(id)sender {
    [AKTools setProperty:conv.dishWellBalance withSlider:(NSSlider *)sender];
}
@end

File: iOS Examples/Swift/Harmonizer/Harmonizer/ViewController.swift
//
//  ViewController.swift
//  Harmonizer
//
//  Created by Nicholas Arner on 10/7/14.
//  Copyright (c) 2014 Aurelius Prochazka. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let harmonizer = HarmonizerInstrument();
    let sampler = AKSampler();
    
    override func viewDidLoad() {
        super.viewDidLoad()
        AKOrchestra.addInstrument(harmonizer)
    }
    
    @IBAction func startRecording(sender: AnyObject) {
        harmonizer.play()
        sampler.startRecordingToTrack("harmonizer")
    }
    
    @IBAction func stopRecording(sender: AnyObject) {
        harmonizer.stop()
        sampler.stopRecordingToTrack("harmonizer")
    }
    
    @IBAction func startPlaying(sender: AnyObject) {
        sampler.startPlayingTrack("harmonizer")
    }
    
    @IBAction func stopPlaying(sender: AnyObject) {
        sampler.stopPlayingTrack("harmonizer")
    }
}